### DISCUSSION TITLE
Username and password not taking while clinong the repositoryi

### DISCUSSION THREAD

**Discussion by partnervarsha** [community/community#129317] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

while i am cloning the git repository it is asking username and password and i have given too but its not taking

**Comment by davevad93** [comment]
Hi @partnervarsha ,

To do git operations such as `git clone` or `git push` authentication is needed, so first of all you need to configure the `global user.name` and `global user.email`:

[Getting Started - First-Time Git Setup](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)

If you're using `HTTPS` you need to use a personal access token as password, you can also use `SSH` for which you would need to generate an ssh key and create a password.

Here's how to create a personal access token:

[Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

And here how to generate an SSH key:

[Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
